### ISSUE TITLE
no-name-in-module for Sequence, Callable, Iterator in Python 3.11 collections.abc

### ISSUE THREAD

**Issue by TylerYep** [PyCQA/pylint#6778]
### Bug description

Importing from collections.abc gives off these false positive errors

### Command used

```shell
pre-commit
```

### Pylint output

```shell
************* Module stats
stats.py:3:0: E0611: No name 'Callable' in module 'collections.abc' (no-name-in-module)
************* Module user
user.py:1:0: E0611: No name 'Sequence' in module 'collections.abc' (no-name-in-module)
```

### Expected behavior

No errors

### Pylint version

```shell
pylint 2.13.9 and 2.14.0-b1
```

### OS / Environment

Mac OSX
Python 3.11.0b1

**Comment by DanielNoord** [comment]
This has been fixed on current `main` of `astroid`. We just need to release a version and bump it in `pylint`.

**Comment by Pierre-Sassoulas** [comment]
There's currently nothing that need backport in astroid : https://github.com/PyCQA/astroid/issues?q=is%3Aopen+is%3Aissue+label%3Aneeds-backport. We were not going to use 2.12 in pylint 2.14 so probably fixed in 2.15 (?) Or we need to backport the fix in astroid 2.11.

**Comment by DanielNoord** [comment]
Let's not backport. 2.11 doesn't explicitly support Python 2.11.

We need all our attention for releasing 2.14 and then 2.12 now.

**Comment by DanielNoord** [comment]
Closing this as we now have basic support for `3.11`.
